### PR TITLE
add webhook scalingGroup support

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/rancher/rancher/issues/8519

### CODE REVIEW

File: drivers/scale_service.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func (s *ScaleServiceDriver) ValidatePayload(conf interface{}, apiClient *client
 		return http.StatusBadRequest, fmt.Errorf("Invalid service %v", config.ServiceID)
 	}
 
-	if service.Kind != "service" && service.Kind != "loadBalancerService" {
+	if service.Kind != "scalingGroup" && service.Kind != "loadBalancerService" {
```
Comment (deniseschannon) [inline drivers/scale_service.go]: @cjellick Would we want to keep backwards compatibility to for `service` vs. `scalingGroup`? With this change, we'd have to branch our webhook service for the 1.6.x line and master.

**Comment by deniseschannon** [comment]
@carolynhu Also, do tests need to change for the fact that we have scalingGroup versus service?

**Comment by carolynhu** [comment]
@deniseschannon no, we do not need to change the tests, since in the scale_service_test, there is no hard code for test "service".

File: drivers/scale_service.go (hunk 1 above)
Comment (cjellick) [inline drivers/scale_service.go]: yep needs b backwards compatible. handle both cases

**Comment by cjellick** [comment]
@carolynhu when you run the validation tests, make sure you run them against a v1.6 and a master setup and that they pass with both

File: drivers/scale_service.go (hunk 1 above)
Comment (carolynhu) [inline drivers/scale_service.go]: yes, I think we can have the two conditions included to make it backward compatible

**Pull request closed without merging**